packages/flutter/test/material/ink_splash_test.dart

```diff
@@ -0,0 +1,29 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  // Regression test for https://github.com/flutter/flutter/issues/21506.
+  testWidgets('InkSplash receives textDirection', (WidgetTester tester) async {
+    await tester.pumpWidget(new MaterialApp(
+      home: new Scaffold(
+      appBar: new AppBar(title: const Text('Button Border Test')),
+      body: new Center(
+        child: new RaisedButton(
+          child: const Text('Test'),
+          onPressed: () {},
+          shape: Border.all(
```

Comment:
It seems weird that `new Border.all` ends up requiring a textDirection. That one should be direction-agnostic.